### PR TITLE
update THIRDPARTY

### PR DESCRIPTION
**Pull request by timosachsenberg**

### CODE REVIEW

**Comment by timosachsenberg** [comment]
```
 Comet version 2019.01 rev. 4

 The comet.params file is from version 2016.01
 Please update your comet.params file.  You can generate
 a new parameters file using "comet -p"
```
ok seems we need to update the param file (or do we create it completely from scatch? Need to check)
also the mods got an extra  double in the last column
variable_mod03 = 0.0 X 0 3 -1 0 0 0.0  (thanks @oliveralka for pointing that out)

**Comment by oliveralka** [comment]
Here the comet params files for comparison:
https://drive.google.com/drive/folders/12VE0gG2WUz-hK6qK8mV_cQ8H2Zgfofx-?usp=sharing

Some of the standard parameter values have been changed (probably set in OpenMS, so it does not matter). 

Additional paramters (new):
```
peff_format = 0  
peff_obo =
search_enzyme2_number = 0              # second enzyme; set t
peptide_length_range = 5 63   
max_duplicate_proteins = 20
equal_I_and_L = 1 
```

Removed parameters:
`output_outfiles = 0`

Not sure if everything is implemented to support the new parameters.

**Comment by jpfeuffer** [comment]
No, we write a fixed version number on top of our autogenerated params file. You need to change it in the code.
See https://github.com/OpenMS/OpenMS/issues/4152

File: src/topp/CometAdapter.cpp [hunk 1]
```diff
@@ -285,12 +295,13 @@ class TOPPCometAdapter :
...
-    os << "num_enzyme_termini = " << num_enzyme_termini[getStringOption_("num_enzyme_termini")] << "\n";                  // 1 (semi-digested), 2 (fully digested, default), 8 C-term unspecific , 9 N-term unspecific
+    os << "search_enzyme2_number = " << enzyme2_number << "\n";              // second enzyme; set to 0 if no second enzyme
+    os << "num_enzyme_termini = " << num_enzyme_termini[getStringOption_("num_enzyme_termini")] << "\n"; // 1 (semi-digested), 2 (fully digested, default), 8 C-term unspecific , 9 N-term unspecific
     os << "allowed_missed_cleavage = " << getIntOption_("allowed_missed_cleavages") << "\n";             // maximum value is 5; for enzyme search
 
     // Up to 9 variable modifications are supported
-    // format:  <mass> <residues> <0=variable/else binary> <max_mods_per_peptide> <term_distance> <n/c-term> <required>
-    //     e.g. 79.966331 STY 0 3 -1 0 0
+    // # format:  <mass> <residues> <0=variable/else binary> <max_mods_per_peptide> <term_distance> <n/c-term> <required> <neutral_loss>
+    //     e.g. 79.966331 STY 0 3 -1 0 0 97.976896
```
Comment (timosachsenberg) [inline src/topp/CometAdapter.cpp]: not sure if we should always add neutral losses

Comment (jpfeuffer) [reply src/topp/CometAdapter.cpp]: but how do we get the possible losses for the different mods? is this registered somewhere?

Comment (timosachsenberg) [reply src/topp/CometAdapter.cpp]: yep in residue db. but we only store one and in reallity it depends on the fragmentation method etc.

File: src/tests/topp/THIRDPARTY/CometAdapter_2_out.idXML [hunk 2]
```diff
@@ -1,105 +1,105 @@
...
+			<ProteinHit id="PH_4" accession="prot5" score="0.0" sequence="" >
 			</ProteinHit>
-			<ProteinHit id="PH_5" accession="prot6" score="0" sequence="" >
+			<ProteinHit id="PH_5" accession="prot6" score="0.0" sequence="" >
 			</ProteinHit>
-			<ProteinHit id="PH_6" accession="prot10" score="0" sequence="" >
+			<ProteinHit id="PH_6" accession="prot10" score="0.0" sequence="" >
 			</ProteinHit>
-			<UserParam type="stringList" name="spectra_data" value="[/Users/builder/jenkins/ws/openms_pr_ci/c797036f/source/src/tests/topp/THIRDPARTY/spectra_comet.mzML]"/>
+			<UserParam type="stringList" name="spectra_data" value="[CometAdapter_2_prepared.mzML]"/>
```
Comment (jpfeuffer) [inline src/tests/topp/THIRDPARTY/CometAdapter_2_out.idXML]: Is this correct? Why did that change?

File: src/topp/CometAdapter.cpp [hunk 3]
```diff
@@ -433,10 +479,14 @@ class TOPPCometAdapter :
     os << "max_precursor_charge = " << getIntOption_("max_precursor_charge") << "\n";                // set maximum precursor charge state to analyze (allowed max 9)
     os << "nucleotide_reading_frame = " << 0 << "\n";            // 0=proteinDB, 1-6, 7=forward three, 8=reverse three, 9=all six
     os << "clip_nterm_methionine = " << (int)(getStringOption_("clip_nterm_methionine")=="true") << "\n";              // 0=leave sequences as-is; 1=also consider sequence w/o N-term methionine
+    os << "peptide_length_range = 5 63\n";                       // minimum and maximum peptide length to analyze (default 1 63; max length 63)
```
Comment (jpfeuffer) [inline src/topp/CometAdapter.cpp]: Could you expose this to the params as well? Other search engines have this parameter. We should sync.

File: src/tests/topp/THIRDPARTY/CometAdapter_2_out.idXML (hunk 2 above)
Comment (timosachsenberg) [reply src/tests/topp/THIRDPARTY/CometAdapter_2_out.idXML]: hmm from this line it seems correct. Though I think it should annotate the absolute path.
 https://github.com/OpenMS/OpenMS/blob/9d2282c40359c6cbd8e13e8bd220c8737b4e377a/src/tests/topp/THIRDPARTY/third_party_tests.cmake#L162

File: src/topp/CometAdapter.cpp (hunk 3 above)
Comment (timosachsenberg) [reply src/topp/CometAdapter.cpp]: will do

File: src/tests/topp/THIRDPARTY/CometAdapter_2_out.idXML (hunk 2 above)
Comment (timosachsenberg) [reply src/tests/topp/THIRDPARTY/CometAdapter_2_out.idXML]: TODO:
in1:  C:\jenkins\ws\openms\PR\bldtst\031d6063\build\src\tests\topp\CometAdapter_1_out1.tmp   (line: 8, position/column: 4/26)
			<!
			 UserParam type="stringList" name="spectra_data" value="[C:/jenkins/ws/openms/PR/bldtst/031d6063/source/src/tests/topp/THIRDPARTY/spectra_comet.mzML]"/>

in2:  C:\jenkins\ws\openms\PR\bldtst\031d6063\source\src\tests\topp\THIRDPARTY\CometAdapter_1_out.idXML   (line: 8, position/column: 4/26)
			<!
			 ProteinHit id="PH_0" accession="test2_rev" score="0.0" sequence="" >

**Comment by timosachsenberg** [comment]
rebuild jenkins

File: src/topp/CometAdapter.cpp (hunk 3 above)
Comment (timosachsenberg) [reply src/topp/CometAdapter.cpp]: in a later PR

**Comment by jpfeuffer** [comment]
We have to check #4614 since this PR seems to put a "-1" into the maxMissedCleavages now. Not sure if this is related.

**Comment by jpfeuffer** [comment]
It sometimes works. I will merge so people with windows have an incentive to fix this. Until then, no nightlies.

**Pull request merged**